### ISSUE TITLE
Feature: Relative Links for Images

### ISSUE THREAD

**Issue by VillageHubertChen** [classeur/classeur#147]
# Current Workflow _without_ Feature
1. When publishing articles on the server we use relative links for images
2. After copying the article from server to Classeur the images are missing in preview
3. Determine base URL to make a global search and replace
4. Make global search and replace
5. Now have pictures in preview. Do the edit
6. On publishing do search and replace again to remove base URL
# Workflow _with_ Feature
1. Copy production Article from Server
2. Copy base URL into matching field in preferences. Now all images show in preview.
# I would write Tutorial

Like in the example of the Zen Mode I would write the tutorial and enter it in our growing "Missing Classeur Manual". I also can do beta testing.

**Comment by orschiro** [comment]
Thanks again for your enhancement request, Hubert! 

What is the exact feature you're requesting? That we support relative links for images?

Thankfully,

~Robert

**Comment by VillageHubertChen** [comment]
Dear Robert,

Thanks for your swift reply. Yes, exactly. When pasting a markdown article from the server (such as GitHub Pages) to Classeur all images with relative links do not show in the preview pane. If you were to add a _Base URL_ field in your preferences you could add the base url prefix to all relative links when translating to HTML. This would make images with relative links render the image content in the preview pane.

Let me know if you have any more questions,

~Hubert

**Comment by orschiro** [comment]
> Base URL field in your preferences you could add the base url prefix to all relative links when translating to HTML

Interesting idea and probably easy to implement by a new optional field in the preferences. 

Waiting to see what @benweet thinks.

**Comment by VillageHubertChen** [comment]
Exactly. And best of all, even though it is easy to implement it is of high value.

**Comment by VillageHubertChen** [comment]
Here is an additional idea: I got the push to GitHub working. ( I am writing on instructions right now ). When you set up your connection to GitHub you provide already: 1) URL to repository 2) Path to markdown file. This is enough information to assemble a working absolute URL for GitHub. As such you can implement this feature transparently, as in: It just works! If you have any questions, let me know.

**Comment by orschiro** [comment]
Thanks Hubert for the addition! Let's wait for Ben to evaluate this request in its entirety. 

~Robert
